xarray/plot/plot.py

```diff
@@ -141,6 +142,9 @@ def plot(darray, row=None, col=None, col_wrap=None, ax=None, hue=None,
     """
     darray = darray.squeeze()
 
+    if isinstance(darray.data, dask_array_type):
+        darray = darray.compute()
```

Comment:
`compute()` works even on numpy arrays, so let's just write `darray = darray.compute()` here without the `if` check.